applications/vanilla/library/MenuCounts/DiscussionCounterProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Forum\Menu;
+
+use Vanilla\Menu\CounterProviderInterface;
+use Vanilla\Menu\Counter;
+
+/**
+ * Menu counter provider for discussion model.
+ */
+class DiscussionCounterProvider implements CounterProviderInterface {
+
+    /** @var \DiscussionModel */
+    private $discussionModel;
+
+    /**
+     * Initialize class with dependencies
+     *
+     * @param \DiscussionModel $discussionModel
+     */
+    public function __construct(\DiscussionModel $discussionModel) {
+        $this->discussionModel = $discussionModel;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getMenuCounters(): array {
+        $counters[] = new Counter("Participated", $this->discussionModel->getCountParticipated());
```

Comment:
Getting the following fatal error from this line.

> The "DiscussionModel" object does not have a "xgetCountParticipated" method.

Reply:
Good catch. Moved to addons repo. Please check: https://github.com/vanilla/addons/pull/697